### DISCUSSION TITLE
How to write a docstring?

### DISCUSSION THREAD

**Discussion by flashgorman-chorus** [pydantic/pydantic#2966] [Question] [answered]
What is the "proper" way to write a docstring for a Pydantic model class?  Like this?

```
class Crayon(Model):
    """Encapsulates the details of a crayon.

    Attributes:
        color: The color of the crayon.
    """
    color: str
```

The reason I ask is that, if I do it the way I _think_ I should do it, my IDE complains that it can't find the attributes in the docstring, so I wondered if there was a different convention I should be following?

**Comment by samuelcolvin** [comment, marked as answer]
Sounds like a problem with your IDE, nothing specific to pydantic AFAIK.

PyCharm (and probably other IDEs) allow you to choose between different formats of docstring.

But you don't really need to worry about formally formatted docstrings since you can use pydantic's schema generation for anything at runtime.
